### DISCUSSION TITLE
Merging a new ODF with existing .cmb

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/grandorgue#1722] [Help]
I am working on a new sample-set and having an issue with the cmb file. The organ was voiced and the cmb file transfered to another computer having the same set and an error comes up that the "file does not exactly match the current odf...."

If I try to import the file a window comes up that asks for either a yaml or cmb file.

What exactly should I be doing? I cannot publish this set because of this. Besides, I need to add the crescendo pedal to the voicing data and use the same for different versions of the same instrument....... same instrument, same switches (ranks), same crescendo pedal but different features especially with regards to pistons.

How do I best get about all this?

Mark

**Comment by Bumblebee001** [comment]
Ideally, the relevant data is made part of the odf somehow and editing possible to adapt / correct accordingly.

**Comment by larspalo** [comment]
To get the .cmb file "universally" usable for other people on other computers you may not do any kind of personal customizations for the organ on the computer you generate the .cmb file from that doesn't exist for any other user. To achieve this you need to open the sample set completely clean on your .cmb generating computer (_without any previously saved_ settings at all). Nor can you do any adjustments to the odf afterwards (without triggering the warning message) when the .cmb is generated.

For your use case you should, from a completely clean opening of the sample set, only program the crescendo without anything else done to it until you've packaged the organ for distribution - only then will it work without warnings.

**Comment by oleg68** [comment]
> What exactly should I be doing?

Import each .cmb or .yaml file and save them again.

Next time GO will not ask you anything.

**Reply by larspalo** [reply]
@oleg68 Yes, that would work, but it's better if the distributor of the .cmb file (together with the corresponding .organ file) creates it without any special custom settings that couldn't match any other users setup so that no warnings pop up at all. This is somewhat tricky but not impossible if one knows what to do.

The crescendo issue is already a separate issue as #1624, and I've recently implemented the import of voicing data from a GO generated .cmb file in GOODF that will merge that information with the .organ file to get a odf file that contain all the combined data for those who are worried about the bundled .cmb and .organ files (or have trouble creating them).

**Reply by oleg68** [reply]
@larspalo if I understood the starting message correctly, @Bumblebee001 has two different questions:
- How to import a preset with voiicing and crescendo to the new ODF
- How to use a lot of combination files without extra asking

You answer mostly the first question, but I answer mostly the second question.
